### PR TITLE
release: 4.2.0

### LINKED ISSUE #4942: release: v4.2.0
## Checklist

- [x] Assign to next release manager
- [ ] Close previous release milestone
- [ ] Create next release milestone
- [ ] Associate issue with next release milestone

### PR DESCRIPTION
**Pull request by jsjoeio**
This PR is to generate a new release of `code-server` at `v4.2.0`

## Screenshot

image

## TODOs

Follow "Publishing a release" steps in `ci/README.md`

- [ ] publish release and merge PR
- [ ] update the AUR package

Fixes #4942

### CODE REVIEW

File: ci/helm-chart/Chart.yaml [hunk 1]
```diff
@@ -20,4 +20,4 @@ version: 2.2.0
 # This is the version number of the application being deployed. This version number should be
 # incremented each time you make changes to the application. Versions are not expected to
 # follow Semantic Versioning. They should reflect the version the application is using.
-appVersion: 4.1.0
+appVersion: 4.2.0
```
Comment (code-asher) [inline ci/helm-chart/Chart.yaml:23]: I think we are supposed to bump `version` in this file as well.

**Review by code-asher — APPROVED** [review]
One spot I think we need to update, otherwise looks good!

File: ci/helm-chart/Chart.yaml (hunk 1 above)
Comment (jsjoeio) [reply ci/helm-chart/Chart.yaml:23]: ah! that must not be part of the release prep script. nice catch!

**Comment by benz0li** [comment]
@jsjoeio Help > About reports `code-server: vUnknown`.

version

**Comment by jsjoeio** [comment]
This looks like a bug: https://github.com/coder/code-server/pull/5014#issuecomment-1076026582

Do you mind opening a bug report for that? You can leave most of it blank (I know how to reproduce).

> @jsjoeio I have noticed, that the Assets for v4.2.0 are ~90 MB smaller than the ones for v4.1.0.
>
> ℹ️ There are no source map files in v4.2.0, although being pointed to in the JS files (e.g. lib/vscode/out/vs/loader.js).
>
> Should I open an issue regarding this?

Great question! I'm not sure if that was intentional 🤔 Could have missed a patch potentially. Any ideas @code-asher ?

**Comment by code-asher** [comment]
Definitely not intentional.  I bet it is due to using Code's packaging task instead of cobbling it together ourselves.  They might be removing the source maps because they host them elsewhere.  So we might have to patch that removal out.

**Comment by jsjoeio** [comment]
Ahhh gotcha. We can track here: https://github.com/coder/code-server/issues/5026

Thanks @benz0li for reporting!

**Pull request merged**